create_train_test_split.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""
+A file for creating the train-test-validation split for the quranic text file.
+
+Author: Hamzah Khan
+Date: Jan. 23, 2019
+"""
+
+from collections import defaultdict
+from sklearn.model_selection import train_test_split
+import utils.text as text_utils
+
+from argparse import ArgumentParser
+
+# This gives us a 60-20-20 split.
+DEFAULT_RANDOM_SEED = 1
+TRAIN_SPLIT_FRACTION = 0.6
+TEST_SPLIT_FRACTION  = 0.2
+VALIDATION_SPLIT_FRACTION = 0.2
+
+parser = ArgumentParser(description='Tarteel Data Train-Test-Validation Splitter')
+parser.add_argument('-i', '--path_to_quran_json', type=str, default="data/data-uthmani.json",
+                    help="Path to the Qur'anic text Json file.")
+parser.add_argument('-o', '--output_path', type=str)
+parser.add_argument('-g', '--group_identical_text', action='store_true',
+                    help="""If True, ayahs with identical text will be grouped into one set, not spread across
+                            multiple sets.""")
+parser.add_argument('--train_frac', type=float, default=TRAIN_SPLIT_FRACTION)
+parser.add_argument('--test_frac', type=float, default=TEST_SPLIT_FRACTION)
+parser.add_argument('--validation_frac', type=float, default=VALIDATION_SPLIT_FRACTION)
+parser.add_argument('-s', '--seed', type=int, default=DEFAULT_RANDOM_SEED)
+parser.add_argument('-v', '--verbose', action='store_false')
+args = parser.parse_args()
+
+# Define constants.
+NUM_SURAHS = 114
+
+def create_train_test_validation_split(ayahs_to_text, train_test_validate_fractions, should_group_identical_text=True, random_seed=DEFAULT_RANDOM_SEED):
+    """
+    Create a train-test-validation split over ayahs with the same text, given the Qur'anic data.
+    Returns a list of lists, each an ayah group, containing the ayah numbers.
+    """
+    train_frac = train_test_validate_fractions[0]
+    test_frac = train_test_validate_fractions[1]
+    validate_frac = train_test_validate_fractions[2]
```

Comment:
Would prefer a small class `ValidationParams` with properties `train/test/validate_fraction`.
This is fine tho.